MobileCenterPush/MobileCenterPush/MSPush.m

```diff
@@ -102,50 +103,45 @@ + (void)resetSharedInstance {
   sharedInstance = nil;
 }
 
-- (void)registerForRemoteNotifications {
+- (void) registerForRemoteNotifications {
   MSLogVerbose([MSPush logTag], @"Registering for push notifications");
 #if !(TARGET_OS_SIMULATOR)
   if (floor(NSFoundationVersionNumber) <= NSFoundationVersionNumber_iOS_9_x_Max) {
-    UIUserNotificationType allNotificationTypes = (UIUserNotificationType)(
-        UIUserNotificationTypeSound | UIUserNotificationTypeAlert | UIUserNotificationTypeBadge);
-    UIUserNotificationSettings *settings =
-        [UIUserNotificationSettings settingsForTypes:allNotificationTypes categories:nil];
+    UIUserNotificationType allNotificationTypes = (UIUserNotificationType) (UIUserNotificationTypeSound | UIUserNotificationTypeAlert | UIUserNotificationTypeBadge);
+    UIUserNotificationSettings *settings = [UIUserNotificationSettings settingsForTypes:allNotificationTypes categories:nil];
     [[UIApplication sharedApplication] registerUserNotificationSettings:settings];
   } else {
     UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
-    UNAuthorizationOptions authOptions =
-        (UNAuthorizationOptions)(UNAuthorizationOptionAlert | UNAuthorizationOptionSound | UNAuthorizationOptionBadge);
+    UNAuthorizationOptions authOptions = (UNAuthorizationOptions) (UNAuthorizationOptionAlert | UNAuthorizationOptionSound | UNAuthorizationOptionBadge);
     [center requestAuthorizationWithOptions:authOptions
-                          completionHandler:^(__attribute__((unused)) BOOL granted,
-                                              __attribute__((unused)) NSError *_Nullable error){
-                          }];
+                          completionHandler:^(BOOL granted, NSError * _Nullable error) {}];
+    
+    [[UNUserNotificationCenter currentNotificationCenter] setDelegate:[[UIApplication sharedApplication] delegate]];
```

Comment:
Do we actually need this as of now? Looks like we don't really need it except if we decide to override the OS behavior and show system notifications while app in foreground for instance? If so we should provide our own delegate for people that doesn't want to bother implementing it. See ME for examples.